config/coordinator.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"crypto/md5"
+	"encoding/binary"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// Coordinator coordinates Alertmanager configurations beyond the lifetime of a
+// single configuration.
+// TODO: Make Coordinator thread safe
+type Coordinator struct {
+	// TODO: Either do all cfg or all conf
+	cfgFilePath string
+	logger      log.Logger
+
+	config *Config
+
+	configHashMetric        prometheus.Gauge
+	configSuccessMetric     prometheus.Gauge
+	configSuccessTimeMetric prometheus.Gauge
+
+	subscribers []func(*Config)
+}
+
+// NewCoordinator returns a new coordinator with the given configuration file
+// path.
```

Comment:
Perhaps mention that the config file is not loaded yet.

Reply:
Good point. Added.